tron/bin/check_tron_jobs.py

```diff
@@ -293,41 +302,110 @@ def guess_realert_every(job):
+        'hour': '%Y.%m.%d-%H',
+        'minute': '%Y.%m.%d-%H.%M',
+        'second': '%Y.%m.%d-%H.%M.%S',
     }
-    kwargs.update(job['monitoring'])
-    if 'realert_every' not in kwargs:
-        kwargs['realert_every'] = guess_realert_every(job)
-    kwargs['check_every'] = "{}s".format(_run_interval)
 
-    status = job["status"]
-    if status == "disabled":
-        kwargs["output"] = "OK: {} is disabled and won't be checked.".format(
+    run_buckets = defaultdict(list)
+    if job_content is not None:
+        # Sort runs by shortdate
+        for run in job_content['runs']:
+            # If interval is invalid, will raise a KeyError
+            shortdate = time.strftime(
+                interval_formats[interval],
+                _timestamp_to_timeobj(run['run_time']),
+            )
+            run_buckets[shortdate].append(run)
+    return dict(run_buckets)
+
+
+def compute_check_result_for_job(client, job):
+    kwargs = m(
+        name="check_tron_job.{}".format(job['name']),
+        source="tron",
+    )
+    if 'realert_every' not in kwargs:
+        kwargs = kwargs.set('realert_every', guess_realert_every(job))
+    kwargs = kwargs.set('check_every', "{}s".format(_run_interval))
+    sensu_kwargs, nonsensu_kwargs = filter_monitoring_config(job['monitoring'])
+    kwargs = kwargs.update(sensu_kwargs)
+
+    kwargs_list = []
+    if job["status"] == "disabled":
+        kwargs = kwargs.set('output', "OK: {} is disabled and won't be checked.".format(
             job['name'],
-        )
-        kwargs["status"] = 0
-        log.info(kwargs["output"])
-        return kwargs
+        ))
+        kwargs = kwargs.set('status', 0)
+        kwargs_list.append(kwargs)
     else:
         # The job is not disabled, therefore we have to look at its run history
         url_index = client.index()
         tron_id = get_object_type_from_identifier(url_index, job["name"])
-        job_content = client.job(
+        job_content = pmap(client.job(
             tron_id.url,
             count=20,
             include_action_runs=True,
-        )
-        results = compute_check_result_for_job_runs(
-            job=job,
-            job_content=job_content,
-            client=client,
-        )
-        kwargs.update(results)
-        log.info(kwargs["output"].split("\n")[0])
-        return kwargs
+        ))
+
+        if _are_runs_precious(nonsensu_kwargs):
+            dated_runs = sort_runs_by_interval(job_content, interval='day')
+        else:
+            dated_runs = {'': job_content['runs']}
+
+        for date, runs in dated_runs.items():
+            results = compute_check_result_for_job_runs(
+                job=job,
+                job_content=job_content.set('runs', runs),
+                client=client,
+            )
+            dated_kwargs = kwargs.update(results)
+            if date:  # if empty date, leave job name alone
+                dated_kwargs = dated_kwargs.set('name', f"{kwargs['name']}-{date}")
```

Comment:
This is an indirect way to do it.

Can you instead set the `name` in some if block where we are more explicitly doing a precious-runs branch?

Reply:
Would it be better if we do this way instead:
    if job['monitoring'].get(PRECIOUS_JOB_ATTR, False):
        dated_runs = sort_runs_by_interval(job_content, interval='day')
        # compute_check_result_for_joj_runs for each run
    else:
        # do the old way

Reply:
Sorry, the format is messed up here. Let me know if you don't understand anything for my comment.

Reply:
I'm not sure it would be better to do it this way because the 'old way' is a trivial case of the 'new way'. In other words, the new way involves bucketing by day, while you can essentially treat the old way as putting everything in one bucket, a bucket of eternity. 

In any event, even if we put it into its own branch, the code would more or less be the same in the branch.